### ISSUE TITLE
Season match inconsistency

### ISSUE THREAD

**Issue by ratoaq2** [Diaoul/subliminal#678]
For the latest version: 2.0.3

There's a season match inconsistency when release names have no season (I'll provide real examples later today). 

Given a video file as bellow:
`Show.Name.E05.720p.x264.mkv`

The created video object will have season 1, since guessit will not return season, but this [code](https://github.com/Diaoul/subliminal/blob/master/subliminal/video.py#L168) will default it to 1.

Then when the subtitles (e.g.: `Show.Name.E05.720p.x264,srt`) are found (by any provider), some/most of them will use guessit to retrieve all the release information and this [code](https://github.com/Diaoul/subliminal/blob/master/subliminal/subtitle.py#L207) will prevent a season match, since `video.season == 1` but `'season' is not in guess`

Either the season shouldn't be defaulted to 1 in the video or the compute score should follow the same logic.

**Comment by Diaoul** [comment]
I agree on that. The code you linked to was introduced to handle mini-series that don't have a season number in the filename. Something more clever is needed here.

**Comment by ratoaq2** [comment]
Just to illustrate and give some real examples, there are series that doesn't follow season/episode numbering on their release names. Not all providers have subtitles for those series.
Here are some examples of episodes without season and/or episode numbers:

- Air by date (guessit: `date`): documentaries episodes or talk shows
`Catalyst.2015.11.10.Einstein.360p.LDTV.WEBRIP.[MPup]`
`Eastenders.2016.05.27.HDTV.x264-DEADPOOL`

- Absolute episode numbering (guessit: `episode`): animes
`Kiznaiver.E06.SUBFRENCH.720p.HDTV.x264-D4KiD`

- Absolute episode using `part` (guessit: `part`):
`The.Night.Of.Part.8.INTERNAL.720p.HDTV.x264`
